src/components/MovieForm/MovieForm.jsx

```diff
@@ -0,0 +1,255 @@
+    let baseUrl = backend.MOVIES;
+    if (method === "PUT") {
+        baseUrl += `${id}/`;
+    }
+    const headers = getHeaders();
+    let res = await fetch(baseUrl, {
+        method: method,
+        headers,
+        body: JSON.stringify(data)
+    });
+    let status = method === "PUT" ? 200 : 201;
+    if (res.status === status) {
+        return await res.json();
+    } else {
+        throw res.status;
+    }
+};
+
+const getMovieDetails = async (id) => {
+    let baseUrl = backend.MOVIES + id + "/";
+    const headers = getHeaders();
+
+    let res = await fetch(baseUrl, {
+        method: "GET",
+        headers
+    });
+
+    if (res.status === 200) {
+        return await res.json();
+    } else {
+        throw res.status;
+    }
+};
+
+const mapResponseData = (source) => ({
+    name: source.movieName,
+    genre: source.movieGenre,
+    director: source.movieDirector,
+    description: source.movieDescription,
+    releaseDate: new Date( source.movieReleaseDate ),
+    image: source?.movieImage || null
+});
+
+const MovieForm = () => {
+    const [data, setData] = useState({
+        name: "",
+        director: "",
+        genre: "",
+        releaseDate: undefined,
+        image: "",
+        description: ""
+    });
+    const [photoLabel, setPhotoLabel] = useState(null);
+    const [method, setMethod] = useState("POST");
+    const [loading, setLoading] = useState(false);
+    const [photoError, setPhotoError] = useState(false);
+    const [error, setError] = useState(false);
+    const [getError, setGetError] = useState(false);
+    const [validated, setValidated] = useState(false);
+    const [resCorrect, setResCorrect] = useState(false);
+    const [disabled, setDisabled] = useState(false);
+
+    let fileInput = React.createRef();
+
+    const location = useParams();
+
+    useEffect(() => {
+        const loadMovie = async () => {
+            setLoading(true);
+            try {
+                let res = await getMovieDetails(location.id);
+                setData(mapResponseData(res));
+                setMethod("PUT");
+            } catch(e) {
+                console.log(e);
+                setGetError(true);
+            } finally {
+                setLoading(false);
+            }
+        };
+        if (location?.id) {
+            loadMovie();
+        }
+    }, []);
+
+    const prepareData = () => ({
+        movieName: data.name,
+        movieDirector: data.director,
+        movieGenre: data.genre,
+        movieReleaseDate: data.releaseDate.toISOString(),
+        movieDescription: data.description,
+        movieImage: data.image
+    });
+
+    const handleSubmit = async (event) => {
+        setError(false);
+        event.preventDefault();
+        const form = event.currentTarget;
+        if (form.checkValidity() === false) {
+            event.stopPropagation();
+        } else {
+            setDisabled(true);
+            try {
+            //    prepareData();
```

Comment:
To można usunąć, bo to nie działało i trzeba było dodać w wierszu poniżej (albo można odkomentować i wtedy wyrzucić z funkcji poniżej, nie wiem jak ładniej)